src/_recursiveMerge.ts

```diff
@@ -0,0 +1,44 @@
+import { uniq } from './uniq';
+import { isCyclic } from './_isCyclic';
+
+export type DeepPartialArray<T> = ReadonlyArray<DeepPartial<T>>;
+// eslint-disable-next-line @typescript-eslint/ban-types
+export type DeepPartial<T> = T extends Function
+  ? T
+  : T extends ReadonlyArray<infer U>
+  ? DeepPartialArray<U>
+  : T extends object
+  ? DeepPartialObject<T>
+  : unknown;
+export type DeepPartialObject<T> = { [P in keyof T]?: DeepPartial<T[P]> };
```

Comment:
1. I think, we should use merge-deep, not partial-deep from type-fest. I added type-fest as a dependency, because merge-deep is too complex
2. It was used in two files. Now it is removed